### PR TITLE
Remove rmdir mvdir support from mongodb

### PR DESCRIPTION
**Pull request by rintcius**
#qz-3741 Done

### CODE REVIEW

**Comment by rintcius** [comment]
@wemrysi - I verified this with @beckyconning but I'll double-check with frontend team.

**Comment by beckyconning** [comment]
Can we still rename collections?

**Comment by beckyconning** [comment]
From reading the source it looks like yes.

File: mongodb/src/main/scala/quasar/physical/mongodb/fs/managefile.scala [hunk 1]
```diff
@@ -56,32 +56,8 @@ object managefile {
     case MoveSemantics.FailIfMissing => RenameSemantics.Overwrite
   }
 
-  def moveDir(src: ADir, dst: ADir, sem: MoveSemantics): MongoFsM[Unit] = {
```
Comment (beckyconning) [inline mongodb/src/main/scala/quasar/physical/mongodb/fs/managefile.scala]: individual collection delete :+1:
individual collection rename :+1:
recursive collection rename :+1:
recursive collection delete :-1:

File: mongodb/src/main/scala/quasar/physical/mongodb/fs/managefile.scala [hunk 2]
```diff
@@ -76,9 +76,7 @@ object managefile {
     if (src === dst)
       ().point[MongoFsM]
     else if (depth(src) == 1)
-      dbNameFromPathM(src) flatMap { dbName =>
-        moveAllUserCollections *> dropDatabase(dbName).liftM[FileSystemErrT]
-      }
+      unsupportedOperation("MongoDb connector does not support moving directories involving databases").raiseError[MongoFsM, Unit]
```
Comment (rintcius) [inline mongodb/src/main/scala/quasar/physical/mongodb/fs/managefile.scala]: @beckyconning @wemrysi - I interpreted that this was the outcome of the conversation between the 2 of you on Slack. Not 100% sure, alternative would be (but that'd make the move unclean IMO): 

```
if (src === dst)
  ().point[MongoFsM]	 
else
  moveAllUserCollections
```

**Comment by rintcius** [comment]
@beckyconning @wemrysi Ready for another look, but not yet ready for merge: 

When testing I noticed that cleanup of tests was also using the functionality that got removed, so I changed that too. There's still a problem in the cleanup of test files: mongo ITs that use `tmp.gen_`-collections *and* result in an error are not cleaned up properly (there are 2 or 3 of such tests among which `s07_join/changedTestStatus`). I'll look into this further tomorrow.

**Comment by rintcius** [comment]
Oh! That cleanup issue already exists in `master`. I have created qz-3753 for that.

**Comment by beckyconning** [comment]
This looks great : ). In the future I think we will be more restrictive but this is perfect for right now.

File: it/src/test/scala/quasar/fs/FileSystemTest.scala [hunk 3]
```diff
@@ -52,10 +54,13 @@ import scalaz.stream.Process
   */
 abstract class FileSystemTest[S[_]](
   val fileSystems: Task[IList[SupportedFs[S]]]
-) extends quasar.Qspec {
+)(implicit Q: QueryFile :<: S, M: ManageFile :<: S) extends quasar.Qspec {
```
Comment (wemrysi) [inline it/src/test/scala/quasar/fs/FileSystemTest.scala]: These constraints should be on `doDelete` as that is the only function that requires them.

File: it/src/test/scala/quasar/fs/FileSystemTest.scala [hunk 4]
```diff
@@ -134,6 +139,29 @@ abstract class FileSystemTest[S[_]](
   def execT[A](run: Run, p: Process[FileSystemErrT[F, ?], A]): FsTask[Unit] =
     p.translate[FsTask](runT(run)).run
 
+  def doDelete(run: Run, dir: ADir): FsTask[Unit] = {
```
Comment (wemrysi) [inline it/src/test/scala/quasar/fs/FileSystemTest.scala]: :sob: I see why this is necessary, but seems so arbitrary. We really need to make loading and cleaning up test data first-class in the test suite.

**Pull request merged**